vignettes/adrs_imwg.Rmd

```diff
@@ -0,0 +1,862 @@
+             -AVAL.confirmed) %>%
+      rename(AVALC = AVALC.confirmed) %>%
+      mutate(AVAL = aval_resp_imwg(AVALC)) %>%
+      mutate(
+        PARAMCD = "COVR",
+        PARAM = "Confirmed Response at Time Point by Investigator",
+        PARCAT1 = "Investigator",
+        PARCAT2 = "IMWG"
+      )
+}
+adrs_imwg <- confirmed_response(adrs) 
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
+  adrs_imwg,
+  display_vars = exprs(USUBJID, PARAMCD, AVISIT, ADT, AVALC)
+)
+```
+
+## Analysis flag derivation {#anlfl}
+
+### Flag One Assessment at Each Analysis Visit (`ANL01FL`)  {#anl01fl}
+
+To confirm responses, we took into account all assessments - 
+including those that took place after a new therapy was started 
+or after progression.
+
+When deriving `ANL01FL` this is an opportunity to exclude any records
+that should not contribute to any downstream parameter derivations.
+
+Common options for `ANL01FL` would be to set null for invalid assessments or
+those occurring after new anti-cancer therapy, or to only flag assessments on or
+after date of first treatment/randomization, or rules to cover the case when a
+patient has multiple observations per visit/date (e.g. by selecting the worst value).
+Another consideration could be extra potential protocol-specific sources of
+Progressive Disease such as radiological assessments, which could be
+pre-processed here to create a PD record to feed downstream derivations.
+
+For the derivation of the parameters it is expected that the subject
+identifier variables (usually `STUDYID` and `USUBJID`) and `ADT` are a
+unique key. 
+
+In the below example we consider only valid assessments and
+those occurring on or after randomization date. If there is more than
+one assessment at a date, the worst one is flagged.
+
+```{r}
+adrs_imwg <- adrs_imwg %>%
+  restrict_derivation(
+    derivation = derive_var_extreme_flag,
+    args = params(
+      by_vars = exprs(!!!get_admiral_option("subject_keys"), ADT),
+      order = exprs(AVAL, RSSEQ),
+      new_var = ANL01FL,
+      mode = "first"
+    ),
+    filter = !is.na(AVAL) & AVALC != "MISSING" & ADT >= RANDDT
+  )
+```
+
+### Exclude Assessments After New Anti-Cancer Therapy (`ANL02FL`) {#anl02fl}
+
+Here is an alternative example where those records occurring after new
+anti-cancer therapy are additionally excluded (where `NACTDT` would be
+pre-derived as first date of new anti-cancer therapy.   
+In our example `NACTDT` is present in `SUPPRS` domain. If not available,
+see `{admiralonco}`
+[Creating and Using New Anti-Cancer Start Date](nactdt.html) for deriving this
+variable).
+
+```{r, eval=TRUE}
+adrs_imwg <- adrs_imwg %>%
+  mutate(
+    ANL02FL = case_when(
+      !is.na(AVAL) & ADT >= RANDDT & ADT < NACTDT ~ "Y",
+      is.na(NACTDT) ~ "Y",
+      TRUE ~ NA_character_
+    )
+  )
+```
+
+### Flag Assessments up to First PD (`ANL03FL`)  {#anl03fl}
+
+To restrict response data up to and including first reported progressive disease
```

Comment:
we can discuss in the call, if this is needed for IMWG for PD anal03fl? A scenario when PD on the 1st visit and then CR on 2nd visit, then CR on the3rd visit, then what will happen?